woocommerce/woocommerce-ios
# [Gift Cards] Fix decoding error when gift card amount has non-integer value

Closes: #9793


## Description

This PR fixes a decoding error for stores using the Gift Cards extension, if the gift card in an order has an amount that is a non-integer. This is because the `amount` property for `OrderGiftCard` was mistakenly given an `Int64` type, so it couldn't handle gift card amounts with decimal values. This is resolved by updating the `amount` to use a `Double` type to handle those non-integer amounts.

### Changes

* Updates the `amount` property in `OrderGiftCard` in the Networking layer to use `Double` type.
* Updates the `amount` attribute in the `OrderGiftCard` entity in the Storage layer use `Double` type. This includes a new Core Data model and migration test for the change.

## Testing instructions

### Prerequisites

1. Install and activate the Gift Cards extension.
2. Create a gift card product.
3. Create an order for a gift card product, to create a working gift card with a code you can apply to an order.
4. Create an order with a non-integer order total, and apply the gift card to pay for the order.

### To test

1. Build and run the app.
2. Open the Orders tab and confirm the order list loads.
3. Select the order from step 4 above and confirm the order details Payment and Gift Card sections show a gift card with non-integer amount applied to the order.

## Screenshots

Before|After - Order list|After - Order details
-|-|-
![Simulator Screenshot - iPhone 14 Pro - 2023-05-22 at 23 41 48](https://github.com/woocommerce/woocommerce-ios/assets/8658164/a0da720e-8dfd-47d8-a7fe-96bb5abd701e)|![Simulator Screenshot - iPhone 14 Pro - 2023-05-22 at 23 42 12](https://github.com/woocommerce/woocommerce-ios/assets/8658164/d749f3a8-c1f8-40fb-bc22-9f9cd72a84ed)|![Simulator Screenshot - iPhone 14 Pro - 2023-05-22 at 23 42 28](https://github.com/woocommerce/woocommerce-ios/assets/8658164/ecd67637-1fbf-4d35-944a-bf06a7d398ea)

---
